### PR TITLE
Add X2InfiltrationMod Templates

### PR DESCRIPTION
**Pull request by cannonfodder1**
Builds successfully with no redscreens anywhere in sight.

### CODE REVIEW

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationModTemplateManager.uc [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+//  WOTCStrategyOverhaul Team
+//---------------------------------------------------------------------------------------
+
+class X2InfiltrationModTemplateManager extends X2DataTemplateManager;
+
+static function X2InfiltrationModTemplateManager GetInfilTemplateManager()
+{
+    return X2InfiltrationModTemplateManager(class'Engine'.static.GetTemplateManager(class'X2InfiltrationModTemplateManager'));
+}
+
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationModTemplateManager.uc]: Add a non-static function `GetModifierForItem(name ItemTemplateName)`

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationMod_DefaultModifiers.uc [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+{
+	local array<X2DataTemplate> Templates;
+	local X2InfiltrationMod Base;
+	Base = new class'X2InfiltrationMod';
+	
+	//Templates.AddItem(Base.static.BuildArmorTemplate(name DataName, int Weight, optional float Mult, optional name MultCat));
+	//Templates.AddItem(Base.static.BuildWeaponTemplate(name DataName, int Weight, optional float Mult, optional name MultCat));
+	//Templates.AddItem(Base.static.BuildCustomTemplate(name DataName, optional int Modifier, optional float Mult, optional name MultCat));
+	
+	Templates.AddItem(Base.static.BuildArmorTemplate('KevlarArmor_InfilMod', 3));
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationMod_DefaultModifiers.uc]: Can't you just do this?

```
Templates.AddItem(BuildArmorTemplate('KevlarArmor_InfilMod', 3));
```

**Comment by Xymanek** [comment]
Two things we should discuss before going forward (i'm not 100% sure myself):

**Config variables**
I don't think we should use `ALLCAPITALS` naming style because it's very hard to read. I think we should keep the normal `CamelCase` just like with other variables. Base game, highlander and other mods use such approach:

* https://github.com/X2CommunityCore/X2WOTCCommunityHighlander/blob/master/X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_CovertAction.uc#L41-L48
* https://github.com/X2CommunityCore/X2WOTCCommunityHighlander/blob/master/X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_CovertAction.uc#L28
* https://github.com/Xymanek/ExtendedPersonnelInfo/blob/master/ExtendedPersonnelInfo/Src/ExtendedPersonnelInfo/Classes/EPI_Section_Bonds.uc#L7-L10

**`X2InfiltrationModTemplate` naming**
We need to define how we name the templates so that we can easily go from item template name to the modifier template. I see 3 possible options:

* `KevlarArmor` -> `KevlarArmor` (just use the item name)
* `KevlarArmor` -> `KevlarArmor_InfilMod` (constant suffix, this is what I see you've chose so far)
* `KevlarArmor` -> `InfilMod_KevlarArmor` (constant prefix)

My personal preference would be 3 (since it's very clear (to me at least) what is the name of the item) but I will be ok with whatever you choose. Regardless, if we use prefix/suffix then that should be inside a `const` variable

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationMod_DefaultModifiers.uc (hunk 2 above)
Comment (cannonfodder1) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationMod_DefaultModifiers.uc]: No, I would like to have the DataSet class contain the functions for ease of access by other mods.

Comment (Xymanek) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationMod_DefaultModifiers.uc]: Yes, I agree. I meant that you can call base class methods without the prefix, just by name

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationModTemplateManager.uc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+function X2InfiltrationModTemplate GetInfilTemplateFromItem(X2ItemTemplate ItemTemplate)
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationModTemplateManager.uc]: Please take the item template name, not the template itself. The reason is that if:

* you have the template then to get the name you just do `Template.DataName`
* you have the name then to get the template you need the manager and then the template from there

**Pull request merged**